engine/collection/ingest/engine.go

```diff
@@ -87,75 +126,122 @@ func New(
 	return e, nil
 }
 
-// Ready returns a ready channel that is closed once the engine has fully
-// started.
+// Start starts the engine by starting the message processing loop.
+func (e *Engine) Start(ctx irrecoverable.SignalerContext) {
+	e.lm.OnStart(func() {
+		e.wg.Add(1)
+		go e.loop(ctx)
+	})
+}
+
+// Ready returns a ready channel that is closed once the engine has fully started.
 func (e *Engine) Ready() <-chan struct{} {
-	return e.unit.Ready()
+	return e.lm.Started()
 }
 
 // Done returns a done channel that is closed once the engine has fully stopped.
 func (e *Engine) Done() <-chan struct{} {
-	return e.unit.Done()
+	e.lm.OnStop()
+	return e.lm.Stopped()
+}
+
+// Process processes the given event from the node with the given origin ID in
+// a blocking manner. It returns the potential processing error when done.
+func (e *Engine) Process(channel network.Channel, originID flow.Identifier, event interface{}) error {
+	err := e.messageHandler.Process(originID, event)
+	if err != nil {
+		if engine.IsIncompatibleInputTypeError(err) {
+			e.log.Warn().Msgf("%v delivered unsupported message %T through %v", originID, event, channel)
+			return nil
+		}
+		return fmt.Errorf("unexpected error while processing engine message: %w", err)
+	}
+	return nil
+}
```

Comment:
As an example, ideally if the context has been cancelled, we would fail this call immediately and not even bother storing the message into the messageHandler.

https://github.com/onflow/flow-go/blob/913abdd0df521bd9c0aba4d0d2a76154716e721d/network/p2p/network.go#L256-L274

shows example of how you might set that up with ComponentManager